### PR TITLE
set "scriptedBatchExecution := false" if sbt 0.13

### LINKED ISSUE #6090: NoSuchMethodException runInParallel when "^^0.13.18;scripted" due to #6054 since sbt 1.4.3
## steps

- https://github.com/xuwei-k/xsbt/commit/e74b3a9a2a71b20e64bd1d210f5a8e475d007507
- https://github.com/xuwei-k/xsbt/runs/1404724499?check_suite_focus=true#step:9:577

run `sbt "scripted project/scripted13"`

```diff
--- a/sbt/src/sbt-test/project/scripted13/build.sbt
+++ b/sbt/src/sbt-test/project/scripted13/build.sbt
@@ -1,5 +1 @@
-lazy val root = (project in file("."))
-  .settings(
-    sbtPlugin := true,
-    resolvers += Resolver.typesafeIvyRepo("releases")
-  )
+enablePlugins(SbtPlugin)
diff --git a/sbt/src/sbt-test/project/scripted13/project/plugins.sbt b/sbt/src/sbt-test/project/scripted13/project/plugins.sbt
index a2a45d929f..217d3d1a59 100644
--- a/sbt/src/sbt-test/project/scripted13/project/plugins.sbt
+++ b/sbt/src/sbt-test/project/scripted13/project/plugins.sbt
@@ -1,5 +1 @@
-libraryDependencies += {
-   "org.scala-sbt" %% "scripted-plugin" % sbtVersion.value
-}
-
 offline := true
diff --git a/sbt/src/sbt-test/project/scripted13/disabled b/sbt/src/sbt-test/project/scripted13/test
similarity index 88%
rename from sbt/src/sbt-test/project/scripted13/disabled
rename to sbt/src/sbt-test/project/scripted13/test
index 7cf13e886d..1617a03a4a 100644
--- a/sbt/src/sbt-test/project/scripted13/disabled
+++ b/sbt/src/sbt-test/project/scripted13/test
@@ -1,7 +1,6 @@
 # This tests that this sbt scripted plugin can launch the previous one
 
-> ++2.10.6
-> ^^0.13.16-M1
+> ^^0.13.18
 
 $ copy-file changes/A.scala src/sbt-test/a/b/A.scala
 > scripted
```

## problem

```
[info] [error] java.lang.NoSuchMethodException: sbt.test.ScriptedTests$.runInParallel(java.io.File, boolean, [Ljava.lang.String;, java.io.File, [Ljava.lang.String;, java.util.List, int)
[info] [error] 	at java.lang.Class.getMethod(Class.java:1786)
[info] [error] 	at sbt.ScriptedPlugin$.$anonfun$scriptedRunTask$1(ScriptedPlugin.scala:115)
[info] [error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[info] [error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[info] [error] 	at sbt.std.Transform$$anon$4.work(Transform.scala:68)
[info] [error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[info] [error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[info] [error] 	at sbt.Execute.work(Execute.scala:291)
[info] [error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[info] [error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[info] [error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[info] [error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info] [error] 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[info] [error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info] [error] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[info] [error] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[info] [error] 	at java.lang.Thread.run(Thread.java:748)
[info] [error] (scriptedRun) java.lang.NoSuchMethodException: sbt.test.ScriptedTests$.runInParallel(java.io.File, boolean, [Ljava.lang.String;, java.io.File, [Ljava.lang.String;, java.util.List, int)
[info] [error] Total time: 14 s, completed Nov 16, 2020 5:39:43 AM
Error:  x project/scripted13 
Error:   Cause of test exception: {line 6}  Command failed: scripted failed
```

## notes

https://github.com/sbt/sbt/pull/6054

### PR DESCRIPTION
**Pull request by xuwei-k**
fix https://github.com/sbt/sbt/issues/6090

### CODE REVIEW

**Review by eed3si9n — DISMISSED** [review]

File: main/src/main/scala/sbt/ScriptedPlugin.scala [hunk 1]
```diff
@@ -78,7 +79,14 @@ object ScriptedPlugin extends AutoPlugin {
...
     scriptedParallelInstances := 1,
-    scriptedBatchExecution := true,
+    scriptedBatchExecution := {
+      CrossVersionUtil.binarySbtVersion(scriptedSbt.value) match {
+        case "0.13" =>
+          false
+        case _ =>
+          true
+      }
+    },
```
Comment (eed3si9n) [inline main/src/main/scala/sbt/ScriptedPlugin.scala]:
```suggestion
    scriptedBatchExecution := CrossVersionUtil.binarySbtVersion(scriptedSbt.value) != "0.13",
```

nit: do we need pattern match?

**Comment by bjaglin** [comment]
This is probably also needed for 1.x, see https://github.com/sbt/sbt/issues/6094

**Comment by eed3si9n** [comment]
I'll merge this as-is for now. Feel free to send a subsequent PRs for address sbt 1.x.

**Review by eed3si9n — APPROVED** [review]

**Comment by mkurz** [comment]
@eed3si9n Playframework also needs the fix for 1.x: https://github.com/playframework/playframework/pull/10537 Thanks...!

**Pull request merged**